### PR TITLE
docs(plugins): Adds docs for wash plugins

### PR DESCRIPTION
**Pull request by thomastaylor312**
This adds a usage guide for the CLI as well as an overview of how plugins work. There is also a guide for plugin developers showing how they can develop plugins (or adapt existing components to be plugins)

### CODE REVIEW

**Comment by thomastaylor312** [comment]
In draft until I finish the developer guide, but wanted to get the first part out there for review

File: docs/ecosystem/wash/plugins.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+title: "Wash plugins"
+date: 2024-04-26T11:02:05+06:00
+draft: false
+sidebar_position: 3
+description: "Using wash plugins"
+---
+
+:::tip
+
+Wash plugins are available starting with version 0.28
```
Comment (ericgregory) [inline docs/ecosystem/wash/plugins.md]: Nit: period at end of sentence here.

File: docs/ecosystem/wash/plugins.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+:::
+
+Wash plugins are a way to extend the functionality of wash. They are Wasm components that can be
```
Comment (ericgregory) [inline docs/ecosystem/wash/plugins.md]: Minor tweak suggestions for this paragraph: 

"Plugins are Wasm components that can extend the functionality of `wash` with new subcommands. This functionality is experimental but enabled by default, meaning we may change the plugin API based on feedback from those using it. The component model will make it very easy to adapt an older version of a plugin to a newer version of the API, with full transparency to users and few if any changes to the plugin code."

File: docs/ecosystem/wash/plugins.md [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+used to add new subcommands to wash. This functionality is experimental, but enabled by default.
+What this means is that we may change the plugin API based on feedback of those using it. However,
+due to the power of the component model, it is very easy to adapt an older version of a plugin to a
+newer version of the API, likely with no changes to the plugin's code and transparent to users.
+
+## Managing and consuming plugins
+
+### Installing a plugin
+
+Installation of a plugin is a relatively straightforward process. A plugin can be installed from a
```
Comment (ericgregory) [inline docs/ecosystem/wash/plugins.md]: Nit: We can probably axe "Installation of a plugin is a relatively straightforward process" and let the process speak for itself.

**Comment by ericgregory** [comment]
Added a few nits and minor suggestions, but everything here looks great so far 🚀

**Review by ericgregory — APPROVED** [review]

**Pull request merged**